rust/agama-cli/src/auth.rs

```diff
@@ -137,12 +154,9 @@ fn read_line_from_file(path: &Path) -> io::Result<String> {
 
 /// Asks user to provide a line of input. Displays a prompt.
 fn read_credential(caption: String) -> io::Result<String> {
-    let mut cred = String::new();
-
-    println!("{}: ", caption);
-
-    io::stdin().read_line(&mut cred)?;
-    if cred.pop().is_none() || cred.is_empty() {
+    let caption = format!("{}: ", caption);
+    let cred = prompt_password(caption.clone()).unwrap();
```

Comment:
In this case, where we have many small functions in this file, I would most probably keep the prefix:

```
rpassword::prompt_password(...)
```